### ISSUE TITLE
Should the coordinator check inputs as far as possible?

### ISSUE THREAD

**Issue by real-or-random** [BlockstreamResearch/bip-frost-dkg#29]
This is related to #9, but only somewhat.

At the moment, the coordinator mostly just relays messages. The verification of the messages is left to the participants, who need to perform it anyway. The coordinator could also check messages. Does this provide any advantages?

**Comment by DarkWindman** [comment]
I agree with @jonasnick that it would increase complexity and lead to duplicated checks. 

Replacing the error handling with a model where only the coordinator can be blamed may seem reasonable, but since the coordinator cannot be fully trusted, this approach becomes problematic. In general case, such a check would be redundant. 
However, as mentioned above, implementers can always add this check at a higher level of the implementation.

**Comment by real-or-random** [comment]
Ok, let's not do it.
